### PR TITLE
Support for building in wasm

### PR DESCRIPTION
**Pull request by aclysma**
Based on #95.

### CODE REVIEW

File: loader/src/packfile_io.rs [hunk 1]
```diff
@@ -269,7 +299,12 @@ impl LoaderIO for PackfileReader {
         }
     }
 
-    fn tick(&mut self, _loader: &mut LoaderState) {}
+    fn tick(&mut self, _loader: &mut LoaderState) {
+        // We require this yield if the runtime is a CurrentThread runtime
```
Comment (aclysma) [inline loader/src/packfile_io.rs:303]: If we use the multithreaded runtime on a machine with only 1 CPU core, will it have the same problem? Or does it have a thread pool separate from the user's code?

File: loader/src/packfile_io.rs [hunk 2]
```diff
@@ -62,6 +67,12 @@ impl Drop for PackfileMessageReaderFile {
     }
 }
 
+#[derive(PartialEq)]
+enum RuntimeType {
```
Comment (aclysma) [inline loader/src/packfile_io.rs:71]: We could expose this if you think it's worthwhile. Or we could remove it and just use #[cfg(not(target_arch = "wasm32"))] directly as needed

File: loader/src/packfile_io.rs (hunk 1 above)
Comment (kabergstrom) [reply loader/src/packfile_io.rs:303]: It'll have a separate thread pool

File: loader/src/packfile_io.rs (hunk 2 above)
Comment (kabergstrom) [reply loader/src/packfile_io.rs:71]: Best to keep the public API minimal for now, until our internals stabilize a bit.

**Comment by kabergstrom** [comment]
LGTM! Thank you

**Pull request merged**